### ISSUE TITLE
Loss of correct line-numbers and other issues

### ISSUE THREAD

**Issue by gregrickaby** [WebDevStudios/wd_s#122]
As of right now, wd_s compilation is broken. SourceMaps and line-numbers don't work. Making it difficult to do our jobs. Debugging on the fly is required, not only to check our work, but the previous developers work too. **We need SourceMaps to efficiently do our jobs.** 

**Introduction**

The old grunt `autoprefixer` was ripping apart the stylesheet causing the SourceMap to break. It's even listed as an issue on their Github. This was the impetus for moving to PostCSS:  _use the official [autoprefixer](https://github.com/postcss/autoprefixer) plugin._

**Quick History of Issues**
- Back in September @efuller noticed that Source Maps and line-numbers weren't lining up. https://github.com/WebDevStudios/wd_s/issues/79
- 19 days ago I merged in PostCSS support (which included the `css-mqpacker` plugin) https://github.com/WebDevStudios/wd_s/commit/a2357e1297e6f7d99c26e68b0a41f88315effbd5
- 6 days later, I ripped out `css-mqpacker` and put `CMQ` back in because `css-mqpacker` was sorting media queries based on a "first come, first sort" type of logic. https://github.com/WebDevStudios/wd_s/commit/cdc5cf8bd89d794cb011d454dc1c6c587bd031ac
- Today I flipped the compilation order of Grunt Watch, and disabled SourceMap inclusion inside `style.min.css`, but the issue persists for `style.css`. https://github.com/WebDevStudios/wd_s/commit/d2a3797208dcbf1666bf99fb51774f702d4eb442

**Down the rabbit hole**
The first iteration of our PostCSS implementation was probably 90% correct, however we had an unintended issue when `css-mqpacker` was unsafely grouping media queries. As a quick fix, I threw the old grunt CMQ plugin back in. 

Nope. Wrong. 

`CMQ` is very basic. It only has one option to disable login. All it does is read strings of text and combine similar strings of text. In doing so, it makes a long stylesheet a little bit shorter. Because of this, it doesn't update the SourceMap. SO, the SourceMap _thinks_ the style sheet is 3,000 lines long - and that's what it tells the web browser (even though it's only 2,948 lines long). Thus, _all the line numbers are off_.

I did some research and found an issue with `css-mqpacker`'s unsafe grouping. Turns out, there is an option to disable the "first come, first sort" order that's [in place by default](https://github.com/hail2u/node-css-mqpacker/blob/master/README.md). 

So I ripped out `CMQ` and placed `css-mqpacker` back in turn off the default sort. I disabled the SourceMap feature in `CSSNano` and it appears to have fixed the issue. 

**tl;dr - Give Us A Solution!**
Add `css-mqpacker` back into the PostCSS grunt task, and pass the option to disable it's default sort order. Disable SourceMaps in CSSNano and just spit out a plain ol' minified file for use on Production.

**Extra Note**
We also need to be mindful about how we write our media queries. I doubt there's any automagic plugin out there that can account for a crazy mix of min/max width queries with different numerical units (e.g.; px, em, rem). You should try to keep media queries simple (min-width) and keep the numeric unit consistent (rem).
